darts/models/forecasting/tft_model.py

```diff
@@ -533,41 +533,48 @@ def __init__(self,
+            Hidden state size of the TFT. It is the main hyper-parameter and common across the internal TFT
             architecture.
         lstm_layers : int
-            number of layers for the Long Short Term Memory (LSTM) Encoder and Decoder (1 is a good default).
+            Number of layers for the Long Short Term Memory (LSTM) Encoder and Decoder (1 is a good default).
         num_attention_heads : int
-            number of attention heads (4 is a good default)
+            Number of attention heads (4 is a good default)
         full_attention : bool
             If `True`, applies multi-head attention query on past (encoder) and future (decoder) parts. Otherwise,
             only queries on future part. Defaults to `False`.
         dropout : float
             Fraction of neurons afected by Dropout.
         hidden_continuous_size : int
-            default for hidden size for processing continuous variables
-        add_cyclic_encoder : optional str
-            If other than None, apply cycling encoding to an attribute of the time index and add it to the
-            `future_covariates`.
-            This allows to use the TFTModel without having to pass future_covariates to `fit()` and `train()`
-            Must be an attribute of `pd.DatetimeIndex`, or `week` / `weekofyear` / `week_of_year` - e.g. "month",
-            "weekday", "day", "hour", "minute", "second". See all available attributes in
-            https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DatetimeIndex.html#pandas.DatetimeIndex.
-            For more information, check out :meth:`datetime_attribute_timeseries()
-            <darts.utils.timeseries_generation.datetime_attribute_timeseries>`
+            Default for hidden size for processing continuous variables
+        add_encoders : optional Dict
+            A large number of past and future covariates can be automatically generated with `add_encoders`.
+            This can be done by adding mutliple pre-defined index encoders and/or custom user-made functions that
+            will be used as index encoders. Additionally, a transformer such as Darts' Scaler() can be added to
+            transform the generated covariates. This happens all under one hood and only needs to be specified at
+            model creation.
+            Read :meth:`SequenceEncoder <darts.utils.data.encoders.SequenceEncoder>` to find out more about
+            `add_encoders`. An example showing some of `add_encoders` features:
+
+                add_encoders={
+                    'cyclic': {'future': ['month']},
+                    'datetime_attribute': {'past': ['hour'], 'future': ['year', 'dayofweek']},
+                    'position': {'past': ['absolute'], 'future': ['relative']},
+                    'custom': {'past': [lambda index: (index.year - 1950) / 50]}
+                    'transformer': Scaler()
+
         add_relative_index : bool
```

Comment:
Hmm, I'm not sure I would add it to `add_encoders` as part of the `SequentialEncoder`. 
This relative index stays constant over all samples and gives the index relative to the sample prediction point.
Whereas the others are computed for the entire train, validation and inference time spans.

I think instead of adding it to the `add_encoders`, we can simply leverage the `FutureIntegerIndexEncoder` in `TFTModel` to compute this `relative_index`

Reply:
OK sounds good